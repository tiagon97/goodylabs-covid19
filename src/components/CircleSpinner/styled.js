import styled, { keyframes } from 'styled-components';

import { Colors } from '../../constants';

const spin = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

export const Spinner = styled.div`
  border-radius: 50%;
  width: ${(props) => props.size}em;
  height: ${(props) => props.size}em;
  font-size: 10px;
  border-top: ${(props) => props.size / 10}em solid rgba(255, 255, 255, 0.2);
  border-right: ${(props) => props.size / 10}em solid rgba(255, 255, 255, 0.2);
  border-bottom: ${(props) => props.size / 10}em solid rgba(255, 255, 255, 0.2);
  border-left: ${(props) => props.size / 10}em solid ${Colors.black};
  animation: ${spin} 1.1s infinite linear;
  margin: 0;
`;
