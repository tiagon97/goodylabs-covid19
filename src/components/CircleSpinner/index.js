import React from 'react';
import PropTypes from 'prop-types';

import { Spinner } from './styled';

const CircleSpinner = ({ size }) => <Spinner size={size} />;

CircleSpinner.propTypes = {
  size: PropTypes.number,
};

CircleSpinner.defaultProps = {
  size: 2,
};

export default CircleSpinner;
