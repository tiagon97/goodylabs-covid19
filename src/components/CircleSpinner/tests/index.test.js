/**
 *
 * Tests for CircleSpinner
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import CircleSpinner from '../index';

describe('<CircleSpinner />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<CircleSpinner />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<CircleSpinner />);
    expect(firstChild).toMatchSnapshot();
  });
});
