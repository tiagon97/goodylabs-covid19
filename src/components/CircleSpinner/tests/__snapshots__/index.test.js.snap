// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleSpinner /> Should render and match the snapshot 1`] = `
<div
  class="sc-AxjAm bzYdQR"
  size="2"
/>
`;
